proxy/logging/LogBuffer.h

```diff
@@ -193,14 +193,15 @@ class LogBuffer
                                   char *alt_printf_str = nullptr);
 
   static void
-  destroy(LogBuffer *lb)
+  destroy(LogBuffer *&lb)
```

Comment:
So, the reference is clearly there so that you don't have to touch nearly as many things. But this code is already pretty gnarly. Adding a reference here means that someone reading a call to "destroy" is very unlikely to guess that the parameter they are passing will be modified. Since it's a delete, though, it's not as though someone who passes the parameter in is expecting to use it usefully afterward.

I can't say that I like it, but it isn't the worst thing.